src/devices/Tm1637/README.md

```diff
@@ -0,0 +1,105 @@
+# TM1637
+
+# Summary
+
+TM1637 is a 6 segments Led controler. It used a 2 wire implementation, one for the clock (CLK), one for the data (DIO). This chip can act as well as a key reader. This part has not been implemented. Only the Led display has been implemented. Most of its modern usages are for 4 to 6 segment displays. 
+
+## Device family
+
+You can find this display as [Grove](http://wiki.seeedstudio.com/Grove-4-Digit_Display/) elements as well as very cheap with no brand. A search on your favorite online shop will give you lots of options. Those simple displays are used a lot for simple clock for example. 
+
+## Usage
+
+You need to create a ```Tm1637``` class with 2 pins, the clock one and the data one. 
+
+```csharp
+Tm1637 tm1637 = new Tm1637(21, 20);
```

Comment:
there's non-zero chance this will work with SPI (DIN=>MISO, MOSI unconnected)

Reply:
Will try once the PR will go thru. Only thing that may not work with SPI is that there is a read of the status so at some point, the state of the DIO pin is changed to input to read the status
```chsarp
// Wait 1 µs, it's the waiting time between clk up and down
// That's according to the documentation
DelayHelper.DelayMicroseconds(ClockWidthMicroseconds, true);

var ack = _controller.Read(_pinDio);
if (ack == PinValue.Low)
{
        _controller.SetPinMode(_pinDio, PinMode.Output);
        _controller.Write(_pinDio, PinValue.Low);
}
```